lib/cylc/tests/test_subprocpool.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python2
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from tempfile import NamedTemporaryFile, TemporaryFile
+import unittest
+
+from cylc.subprocctx import SubProcContext
+from cylc.subprocpool import SubProcPool
+
+
+class TestSubProcPool(unittest.TestCase):
+
+    def test_run_command_basic(self):
```

Comment:
I know I've gone a bit pytest mad, but you could replace this test with
```python
TEST_VALUES = [['truth' , ['true'] , '', '', 0],
               ['lies'  , ['false'], '', '', 1],
               ['parrot', ['echo', 'pirate', 'urrrr'], '', 'pirate urrrr\n', 0]]

@pytest.mark.parametrize('name, args, err, out, ret_code', TEST_VALUES)
def test_run_command_basic_pytest(name, args, err, out, ret_code):
    ctx = SubProcContext(name, args)
    SubProcPool.run_command(ctx)
    assert ctx.err == err
    assert ctx.out == out
    assert ctx.ret_code == ret_code
```

I'm of the opinion that each of the cases in the test should have a separate name, even if you do decide to leave them as-is.

Reply:
I have not converted to pytest, but have broken the tests into separate methods.